swan-lake/resources/featured-scenarios/create-your-first-connector-with-ballerina.md

```diff
@@ -0,0 +1,170 @@
+3. A GitHub account, and Git installed locally.
+4. Visual Studio Code with the [Ballerina extension](https://marketplace.visualstudio.com/items?itemName=WSO2.ballerina) installed.
+
+## Step 1: Set the project structure
+
+1. Create a new GitHub repository with an appropriate name. For Ballerina official connectors, the repository name follows the pattern: `module-ballerinax-<connector-name>` (e.g., `module-ballerinax-twitter`). But for custom connectors, you can choose a name that suits your connector.
+
+2. Clone your newly created repository to your local machine:
+
+   ```
+   git clone https://github.com/<your-username>/<connector-repo-name>.git
+   cd <connector-repo-name>
+   ```
+
+3. Visit the [Ballerina generated connector template on GitHub](https://github.com/ballerina-platform/ballerina-library/tree/main/library-templates/generated-connector-template/files) and copy the entire project structure and content to your local repository folder, making sure to include all files and directories.
+
+4. Your local project structure should now look similar to this:
+
+   ```
+   module-ballerinax-myconnector/
+   ├── .github/
+   ├── ballerina/
+   |   ├── tests/
+   │   ├── Ballerina.toml
+   │   ├── Module.md
+   │   ├── Package.md
+   │   ├── build.gradle
+   │   └── client.bal
+   ├── build-config/
+   ├── docs/
+   │   └── spec/
+   │       └── sanitations.md
+   ├── examples/
+   │   ├── README.md
+   │   ├── build.gradle
+   │   └── build.sh
+   ├── gradle/
+   ├── .gitignore
+   ├── LICENSE
+   ├── README.md
+   ├── build.gradle
+   ├── gradle.properties
+   ├── gradlew
+   ├── gradlew.bat
+   └── settings.gradle
+   ```
+
+> **Tip**: The template includes placeholders for various fields. To update them with your connector-specific metadata, use the provided [Ballerina script](https://github.com/ballerina-platform/ballerina-library/blob/main/library-templates/generated-connector-template/scripts/replace_placeholders.bal).
+
+Detailed information on the Ballerina connector structure can be found in the [Ballerina module contribution guide](https://github.com/ballerina-platform/ballerina-library/blob/main/docs/adding-a-new-ballerina-module.md#directory-structure).
+
+## Step 2: Prepare the OpenAPI Specification
+
+1. Find the OpenAPI specification for the API you want to create a connector for. This is usually available in the API documentation.
+   Example: For Twitter, you can get the latest API specification from the [Twitter OpenAPI endpoint](https://api.twitter.com/2/openapi.json).
+
+2. Save the file as `spec.yaml` (or `spec.json`) in the `docs/spec` directory of your project.
+
+3. You may need to sanitize the OpenAPI specification to ensure compatibility with the Ballerina OpenAPI tool. This process may involve:
+   - adding, removing, or modifying security schemes to customize authentication options.
+   - redefining inline schemas as named schemas to avoid duplication and improve readability.
```

Comment:
Shall we have something like this here?
"After the sanitization, the changes should be added to the `sanitations.md` file"

Reply:
+1. updated in https://github.com/ballerina-platform/ballerina-dev-website/pull/9434/commits/6dc1f93b6ef0158c9cec6637e341db136b2c5385